mbuild/tests/test_utils.py

```diff
@@ -186,3 +188,182 @@ def test_has_ipython(self):
+            (2.10, -8.22, 1.22, 7.20, -2.3, 0),
+            (-12, 1.24, -22.816, 11.28654, 22.28946, 0),
+            (-0.3936, -8.283, 3.183, -3.28, 8.7736, 0),
+            (-8.7, 0, 0, 33.12, -24.42, 0),
+            (-12, 1.393, -1.234, 0.2323, 11.6087, 0),
+            (10.22, 3.293, -34.32, 2.596, 18.211, 0),
+            (3.28629, 7.44211, 1.85995, -14.67569, 2.08734, 0),
+            (5.77183, -2.67148, 0.95814, -4.05848, -0.00001, 0),
+            (5.77183, -2.67148, 0.95814, -4.05848, 20, 0),
+            (10.1, 10.1, 10.1, 10.1, 10.1, 0),
+            (-5.5, -5.5, -5.5, -5.5, -5.5, 0),
+        ],
+    )
+    def test_RB_to_OPLS_and_back_random_values(self, c0, c1, c2, c3, c4, c5):
+        # However, this may not be true for real dihedrals.
+        test_error_tol = 1e-10
+
+        opls_coeffs = RB_to_OPLS(
+            c0, c1, c2, c3, c4, c5, value_error_out_of_tol=False
+        )
+        reversed_RB_coeffs = OPLS_to_RB(
+            opls_coeffs[0],
+            opls_coeffs[1],
+            opls_coeffs[2],
+            opls_coeffs[3],
+            opls_coeffs[4],
+        )
+
+        assert np.all(
+            np.isclose(c0, reversed_RB_coeffs[0], atol=test_error_tol, rtol=0)
+        )
+        assert np.all(
+            np.isclose(c1, reversed_RB_coeffs[1], atol=test_error_tol, rtol=0)
+        )
+        assert np.all(
+            np.isclose(c2, reversed_RB_coeffs[2], atol=test_error_tol, rtol=0)
+        )
+        assert np.all(
+            np.isclose(c3, reversed_RB_coeffs[3], atol=test_error_tol, rtol=0)
+        )
+        assert np.all(
+            np.isclose(c4, reversed_RB_coeffs[4], atol=test_error_tol, rtol=0)
+        )
+        assert np.all(
+            np.isclose(c5, reversed_RB_coeffs[5], atol=test_error_tol, rtol=0)
+        )
+
+    def test_OPLS_to_RB_error_tol_not_float(self):
+        with pytest.raises(
+            TypeError,
+            match=r"The error_tol variable must be a float.",
+        ):
+            f0 = 0.1
+            f1 = 0.1
+            f2 = -0.2
+            f3 = -0.1
+            f4 = 0.2
+            OPLS_to_RB(f0, f1, f2, f3, f4, error_tol="s")
+
+    def test_OPLS_to_RB_f0_is_zero(self):
+        # should throw a warning that f0 is zero
+        text_for_error_tol = (
+            "WARNING: The f0/2 term is the constant for the OPLS dihedral equation, "
+            "which is added to a constant for the RB torsions equation via the c0 coefficient. "
+            "The f0 term is zero in the OPLS dihedral form or is force set to zero in this equation, "
+            "so the dihedral is may not an exact conversion; "
+            "since this constant does not contribute to the force equation, "
+            "this should provide matching results for MD, but the energy for each"
+            "dihedral will be shifted by the real f0/2 value."
+        )
+        with pytest.warns(UserWarning):
+            warn(f"{text_for_error_tol}", UserWarning)
```

Comment:
similar to above comment--line 363 prevents testing whether warning is raised by function (line 369)

Reply:
Fixed

Reply:
However this is not working and I don't know why